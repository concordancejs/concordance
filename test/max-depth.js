const test = require('ava')
const { format, diff } = require('..')

const deep = {
  one: {
    two: {
      three: {
        four: {
          five: 'fin',
        },
      },
      arr: [],
      arr2: [[]],
      date: new Date('1969-07-20T20:17:40Z'),
      date2: Object.assign(new Date('1969-07-20T20:17:40Z'), { foo: 'bar' }),
      error: new Error(),
      fn () {},
      fn2: Object.assign(() => {}, { foo: 'bar' }),
      map: new Map(),
      map2: new Map([['foo', 'bar']]),
      obj: {},
      regexp: /foo/,
      regexp2: Object.assign(/foo/, { foo: 'bar' }),
      set: new Set(),
      set2: new Set(['foo']),
    },
    arr: [{ three: { four: {} } }],
    map: new Map([[{ three: { four: {} } }, { three: { four: {} } }]]),
    set: new Set([{ three: { four: {} } }]),
  },
}

test('format() respects maxDepth option', t => {
  t.snapshot(format(deep, { maxDepth: 3 }))
})

test('diff() respects maxDepth option for equal parts', t => {
  t.snapshot(diff(Object.assign({ unequal: 1 }, deep), Object.assign({ unequal: 2 }, deep), { maxDepth: 3 }))
  t.snapshot(diff({ one: { two: { three: 'baz', three2: ['quux'] } } }, { one: { two: { three: 'qux', three2: ['quux'] } } }, { maxDepth: 1 })) // eslint-disable-line max-len
})

test('properties with increased indentation respect the maxDepth when formatted', t => {
  t.snapshot(format({
    foo: {
      bar: {},
    },
  }, {
    maxDepth: 1,
    theme: {
      property: {
        increaseValueIndent: true,
      },
    },
  }))
})
