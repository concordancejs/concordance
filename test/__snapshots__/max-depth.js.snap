// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diff() respects maxDepth option for equal parts 1`] = `
"  Object {
    one: Object {
      arr: Array [
        Object { … },
      ],
      map: Map {
        Object { … } => Object { … },
      },
      set: Set {
        Object { … },
      },
      two: Object {
        arr: Array [],
        arr2: Array [ … ],
        date: Date 1969-07-20T20:17:40.000Z {},
        date2: Date 1969-07-20T20:17:40.000Z { … },
        error: Error { … },
        fn: Function fn {},
        fn2: Function { … },
        map: Map {},
        map2: Map { … },
        obj: Object {},
        regexp: /foo/,
        regexp2: RegExp { /foo/ … },
        set: Set {},
        set2: Set { … },
        three: Object { … },
      },
    },
-   unequal: 1,
+   unequal: 2,
  }"
`;

exports[`diff() respects maxDepth option for equal parts 2`] = `
"  Object {
    one: Object {
      two: Object {
-       three: 'baz',
+       three: 'qux',
        three2: Array [ … ],
      },
    },
  }"
`;

exports[`format() respects maxDepth option 1`] = `
"Object {
  one: Object {
    arr: Array [
      Object { … },
    ],
    map: Map {
      Object { … } => Object { … },
    },
    set: Set {
      Object { … },
    },
    two: Object {
      arr: Array [],
      arr2: Array [ … ],
      date: Date 1969-07-20T20:17:40.000Z {},
      date2: Date 1969-07-20T20:17:40.000Z { … },
      error: Error { … },
      fn: Function fn {},
      fn2: Function { … },
      map: Map {},
      map2: Map { … },
      obj: Object {},
      regexp: /foo/,
      regexp2: RegExp { /foo/ … },
      set: Set {},
      set2: Set { … },
      three: Object { … },
    },
  },
}"
`;
